designs/2023-only-run-reporting-rules/README.md

```diff
@@ -0,0 +1,150 @@
+`eslint-helpers.js` should also perform a validation check that the `ruleFilter` option is a function.
+
+The new `ruleFilter` function when implemented would look like this, using the `--quiet` flag
+rules outlined in this RFC as an example:
+
+```typescript
+linter.verifyAndFix(text, configs, {
+  ruleFilter: ({ ruleId: string, severity: number }) => {
+    return severity === 2;
+  },
+});
+```
+
+The function acts as a filter on the rules to enable, where removal from the list disables a rule.
+This function would return true for all rules which have a severity of 2 (error), effectively filtering out
+all other rules.
+
+The check for unused `eslint-disable` directives (`--report-unused-disable-directives`)
+should continue to mark `warn` rules as used, even when running with `--quiet`. As the
+rules are not actually run, an assumption would have to be made that all directives
+on rules present only in the un-filtered list are used when in this mode. This is a reasonable
+assumption, as the user likely does not expect `warn` flags to be touched at all in this mode.
+This would also apply to blanket `eslint-disable` directives that disable all rules, which
+should always be assumed to be used when the filter function has not changed the length of the list.
+
+In cases of conflicting flags such as the `--max-warnings` flag, this altered `--quiet` flag
+behavior should be disabled while it is in use. This is to ensure that the `--max-warnings`
+flag continues to work as expected. The filter API should not be set, and existing
+behavior of filtering the output should be used. This altered behavior should be documented
+to clarify that it causes warnings to be run despite the `--quiet` flag.
+
+## Documentation
+
+Both the behaviour modification of impacted flags and the API would make sense to document.
+As they are relatively simple additions, they would be documented inline with the rest
+of the CLI and API documentation. Due to the potentially major performance improvements for
+many setups, I feel mentioning it in the release blog post, alongside the potential benefits,
+would be a good idea to spread awareness.
+
+## Drawbacks
+
+This change modifies a command line flag, as well as a new API method. This adds further
+maintenance burden. However, the implementation is relatively simple, and the benefits
+are significant.
+
+As this flag has many interactions with other systems such as `--max-warnings`, further
+maintenance overhead is introduced by having to ensure it behaves correctly between
+the different flags. This would require updating documentation across all flags and relevant
+settings.
+
+## Backwards Compatibility Analysis
+
+This is a breaking change as it alters the behaviour of the `--quiet` flag.
+While the alterations to the quiet flag should not affect the actual outcome of the command,
+as cases where it would are covered by this RFC, it is still worth noting as the behaviour
+is changing.
```

Comment:
I'd add here the case we discussed in https://github.com/eslint/rfcs/pull/104#discussion_r1071597852. Rules with side effects, such as the `react/jsx-uses-vars` rule, will not be run if set to `"warn"`, so users will need to update their configs to `"error"` after we make this change.

Reply:
It’d be nice if there was a way for a rule to know its own severity, so that this rule could warn when it’s on warn :-)

Reply:
Thanks, I've just added this to the document

Reply:
Looks good, thanks!